api/src/db/db.service.ts

```diff
@@ -250,64 +262,136 @@ export class DbService {
+            };
+            pList.push(this.db.find(query_memberOf_per_type));
+
+            // Include the (group) document itself if the "group" type is included in the options
+            if (options.types.includes(DocType.Group)) {
+                // Use two different queries to allow for effective indexing
+                let query_groupDoc;
+                if (options.from === 0) {
+                    query_groupDoc = {
+                        selector: {
+                            $and: [
+                                {
+                                    type: DocType.Group,
+                                },
+                                {
+                                    _id: {
+                                        $in: options.groups,
                                     },
-                                    {
-                                        // Include the (group) document itself
-                                        _id: {
-                                            $in: options.groups,
-                                        },
+                                },
+                            ],
+                        },
+                        use_index: "type-id-index",
+                    };
+                } else {
+                    query_groupDoc = {
+                        selector: {
+                            $and: [
+                                {
+                                    updatedTimeUtc: {
+                                        $gte: options.from,
                                     },
-                                ],
-                            },
-                        ],
-                    },
-                ],
-            },
-        };
-        return this.db.find(query);
+                                },
+                                {
+                                    type: DocType.Group,
+                                },
+
+                                {
+                                    _id: {
+                                        $in: options.groups,
+                                    },
+                                },
+                            ],
+                        },
+                        use_index: "updatedTimeUtc-type-id-index",
+                    };
+                }
+
+                pList.push(this.db.find(query_groupDoc));
+            }
+
+            // Include the user document
+            if (userId) {
+                pList.push(this.getDoc(userId));
+            }
+
+            Promise.all(pList)
+                .then((res) => {
+                    const docs = res.flatMap((r) => r.docs);
+                    const warnings = res.flatMap((r) => r.warning).filter((w) => w);
+                    if (warnings.length > 0) {
+                        resolve({ docs, warnings });
+                    } else {
+                        resolve({ docs });
+                    }
```

Comment:
Suggestion: use a ternary to avoid the if-else condition.

```suggestion
                        resolve({
                            docs, 
                            warnings: warnings.length > 0 ? warnings : undefined,
                         });

```